fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -3733,10 +3734,78 @@ public void replayRenameRollup(TableInfo tableInfo) {
+        }
+        Column currentColumn = olapTable.getColumn(newColName);
+        if (currentColumn != null) {
+            ErrorReport.reportSemanticException(ErrorCode.ERR_DUP_FIELDNAME, newColName);
+        }
+
+        Map<String, Column> nameToColumn = table.getNameToColumn();
+        nameToColumn.remove(colName);
+        column.renameColumn(newColName);
+        nameToColumn.put(newColName, column);
+
+        DistributionInfo distributionInfo = olapTable.getDefaultDistributionInfo();
+        if (distributionInfo.getType() == DistributionInfo.DistributionInfoType.HASH) {
+            HashDistributionInfo hashDistributionInfo = (HashDistributionInfo) distributionInfo;
+            List<Column> distributionColumns = hashDistributionInfo.getDistributionColumns();
+            for (Column distributionColumn : distributionColumns) {
+                if (distributionColumn.getName().equalsIgnoreCase(colName)) {
+                    distributionColumn.renameColumn(newColName);
+                }
+            }
+        }
+
+        PartitionInfo partitionInfo = olapTable.getPartitionInfo();
+        if (partitionInfo.isRangePartition()) {
+            RangePartitionInfo rangePartitionInfo = (RangePartitionInfo) partitionInfo;
+            List<Column> partitionColumns = rangePartitionInfo.getPartitionColumns();
+            for (Column partitionColumn : partitionColumns) {
+                if (partitionColumn.getName().equalsIgnoreCase(colName)) {
+                    partitionColumn.renameColumn(newColName);
+                }
+            }
+            // for expression range partition will also modify the inner slotRef
+            if (partitionInfo instanceof ExpressionRangePartitionInfo) {
+                ExpressionRangePartitionInfo expressionRangePartitionInfo = (ExpressionRangePartitionInfo) rangePartitionInfo;
+                Expr expr = expressionRangePartitionInfo.getPartitionExprs().get(0);
+                AnalyzerUtils.renameSlotRef(expr, newColName);
+            } else if (partitionInfo instanceof ExpressionRangePartitionInfoV2) {
+                ExpressionRangePartitionInfoV2 expressionRangePartitionInfo = (ExpressionRangePartitionInfoV2) rangePartitionInfo;
+                Expr expr = expressionRangePartitionInfo.getPartitionExprs().get(0);
+                AnalyzerUtils.renameSlotRef(expr, newColName);
+            }
+        } else if (partitionInfo instanceof ListPartitionInfo) {
+            ListPartitionInfo rangePartitionInfo = (ListPartitionInfo) partitionInfo;
+            List<Column> partitionColumns = rangePartitionInfo.getPartitionColumns();
+            for (Column partitionColumn : partitionColumns) {
+                if (partitionColumn.getName().equalsIgnoreCase(colName)) {
+                    partitionColumn.renameColumn(newColName);
+                }
+            }
+        }
+
+        // TODO: log
+        // GlobalStateMgr.getCurrentState().getEditLog().logColumnRename();
```

Comment:
this is necessary, consider to persist meta changes

Reply:
The persistence-related code will be implemented in the next PR, because it is easier to test if persistence is not implemented now.